sympy/solvers/solvers.py

```diff
@@ -1648,11 +1559,8 @@ def _as_base_q(x):
                     # if no Functions left, we can proceed with usual solve
                     if not ftry.has(symbol):
                         cv_sols = _solve(ftry, t, **flags)
-                        cv_inv = _solve(t - f1, symbol, **flags)[0]
-                        sols = list()
-                        for sol in cv_sols:
-                            sols.append(cv_inv.subs(t, sol))
-                        result = list(ordered(sols))
+                        cv_inv = _vsolve(t - f1, symbol, **flags)[0]
+                        result = [{symbol: cv_inv.subs(sol)} for sol in cv_sols]
```

Comment:
Here `_vsolve` returns a list `[acos(_t), -acos(_t) + 2*pi]` but the ordering of this list is nondeterministic. The line above arbitrarily selects the first element of this list.